source/tests/system/nirfmxnr_driver_api_tests.cpp

```diff
@@ -722,6 +715,98 @@ TEST_F(NiRFmxNRDriverApiTests, TxpContiguousMultiCarrierFromExample_FetchData_Da
+  EXPECT_SUCCESS(session, client::set_attribute_i32(stub(), session, "", NIRFMXNR_ATTRIBUTE_MODACC_FREQUENCY_ERROR_ESTIMATION, NIRFMXNR_INT32_MODACC_FREQUENCY_ERROR_ESTIMATION_DISABLED));
+  EXPECT_SUCCESS(session, client::set_attribute_i32(stub(), session, "", NIRFMXNR_ATTRIBUTE_MODACC_SYMBOL_CLOCK_ERROR_ESTIMATION_ENABLED, NIRFMXNR_INT32_MODACC_SYMBOL_CLOCK_ERROR_ESTIMATION_ENABLED_FALSE));
+  EXPECT_SUCCESS(session, client::set_attribute_i32(stub(), session, "", NIRFMXNR_ATTRIBUTE_MODACC_PHASE_TRACKING_ENABLED, NIRFMXNR_INT32_MODACC_PHASE_TRACKING_ENABLED_FALSE));
+  EXPECT_SUCCESS(session, client::set_attribute_i32(stub(), session, "", NIRFMXNR_ATTRIBUTE_MODACC_TIMING_TRACKING_ENABLED, NIRFMXNR_INT32_MODACC_TIMING_TRACKING_ENABLED_FALSE));
+  EXPECT_SUCCESS(session, client::set_attribute_i32(stub(), session, "", NIRFMXNR_ATTRIBUTE_MODACC_IQ_ORIGIN_OFFSET_ESTIMATION_ENABLED, NIRFMXNR_INT32_MODACC_IQ_ORIGIN_OFFSET_ESTIMATION_ENABLED_FALSE));
+  EXPECT_SUCCESS(session, client::set_attribute_i32(stub(), session, "", NIRFMXNR_ATTRIBUTE_MODACC_EVM_REFERENCE_DATA_SYMBOLS_MODE, NIRFMXNR_INT32_MODACC_EVM_REFERENCE_DATA_SYMBOLS_MODE_REFERENCE_WAVEFORM));
+
+  // READ TDMS File
+  auto waveform = load_test_waveform_data<float, nidevice_grpc::NIComplexNumberF32>("NR_FR2_UL_SISO_CC-1_BW-50MHz_SCS-120kHz.json");
+  // Split waveform data in two so it can fit within message limits.
+  std::vector<nidevice_grpc::NIComplexNumberF32> waveform_data_first_half(waveform.data.begin(), waveform.data.begin() + 300000);
+  std::vector<nidevice_grpc::NIComplexNumberF32> waveform_data_second_half(waveform.data.begin() + 300000, waveform.data.begin() + 600000);
+  EXPECT_SUCCESS(session, client::mod_acc_cfg_reference_waveform(stub(), session, "", waveform.t0, waveform.dt, waveform_data_first_half));
+  EXPECT_SUCCESS(session, client::initiate(stub(), session, "", ""));
+  float64 composite_rms_evm_mean_first = GET_ATTR_F64(session, "", NIRFMXNR_ATTRIBUTE_MODACC_RESULTS_COMPOSITE_RMS_EVM_MEAN);
+  float64 in_band_emission_margin_first = GET_ATTR_F64(session, "", NIRFMXNR_ATTRIBUTE_MODACC_RESULTS_IN_BAND_EMISSION_MARGIN);
+  EXPECT_SUCCESS(session, client::mod_acc_cfg_reference_waveform(stub(), session, "", waveform.t0, waveform.dt, waveform_data_second_half));
+  EXPECT_SUCCESS(session, client::initiate(stub(), session, "", ""));
+  float64 composite_rms_evm_mean_second = GET_ATTR_F64(session, "", NIRFMXNR_ATTRIBUTE_MODACC_RESULTS_COMPOSITE_RMS_EVM_MEAN);
+  float64 in_band_emission_margin_second = GET_ATTR_F64(session, "", NIRFMXNR_ATTRIBUTE_MODACC_RESULTS_IN_BAND_EMISSION_MARGIN);
+
+  EXPECT_LT(0.0, composite_rms_evm_mean_first);
+  EXPECT_LT(0.0, composite_rms_evm_mean_second);
+  EXPECT_NE(composite_rms_evm_mean_first, composite_rms_evm_mean_second);
+  EXPECT_TRUE(isnan(in_band_emission_margin_first));
```

Comment:
I looked it up in the CHM and apparently "The margin is not returned in case of clustered PUSCH allocation, or when there is full allocation of resource blocks". We're doing the clustered PUSCH allocation so it made sense. Removed getting that attribute and added more asserts on PUSCH attributes.

Reply:
Interesting! Glad there was a reason behind it.